packages/e2e-tests/specs/performance/site-editor.test.js

```diff
@@ -75,112 +77,123 @@ describe( 'Site Editor Performance', () => {
 		}, html );
 		await saveDraft();
 
-		id = await page.evaluate( () =>
+		postId = await page.evaluate( () =>
 			new URL( document.location ).searchParams.get( 'post' )
 		);
 	} );
 
 	afterAll( async () => {
+		const resultsFilename = basename( __filename, '.js' ) + '.results.json';
+		writeFileSync(
+			join( __dirname, resultsFilename ),
+			JSON.stringify( results, null, 2 )
+		);
+
 		await deleteAllTemplates( 'wp_template' );
 		await deleteAllTemplates( 'wp_template_part' );
 		await activateTheme( 'twentytwentyone' );
 	} );
 
-	beforeEach( async () => {
-		await visitSiteEditor( {
-			postId: id,
-			postType: 'page',
-		} );
-	} );
-
-	it( 'Loading', async () => {
-		const editorURL = await page.url();
-
-		// Number of sample measurements to take.
+	describe( 'Loading', () => {
+		// Number of measurements to take.
 		const samples = 3;
 		// Number of throwaway measurements to perform before recording samples.
-		// Having at least one helps ensure that caching quirks don't manifest in
-		// the results.
+		// Having at least one helps ensure that caching quirks don't manifest
+		// in the results.
 		const throwaway = 1;
+		const iterations = Array.from(
+			{ length: samples + throwaway },
+			( _, i ) => i + 1
+		);
 
-		let i = throwaway + samples;
-
-		// Measuring loading time.
-		while ( i-- ) {
-			await page.close();
-			page = await browser.newPage();
+		it.each( iterations )(
+			`trace large post loading durations (%i of ${ iterations.length })`,
+			async ( i ) => {
+				// Open the test page in Site Editor.
+				await visitSiteEditor( {
+					postId,
+					postType: 'page',
+					path: '/navigation/single',
+				} );
+
+				// Wait for the canvas to become editable.
+				await canvas().waitForSelector(
+					'[data-rich-text-placeholder="Write site tagline…"]'
+				);
+
+				if ( i > throwaway ) {
+					const {
+						serverResponse,
+						firstPaint,
+						domContentLoaded,
+						loaded,
+						firstContentfulPaint,
+						firstBlock,
+					} = await getLoadingDurations();
+
+					results.serverResponse.push( serverResponse );
+					results.firstPaint.push( firstPaint );
+					results.domContentLoaded.push( domContentLoaded );
+					results.loaded.push( loaded );
+					results.firstContentfulPaint.push( firstContentfulPaint );
+					results.firstBlock.push( firstBlock );
+				}
 
-			await page.goto( editorURL );
-			await page.waitForSelector( '.edit-site-visual-editor', {
-				timeout: 120000,
-			} );
-			await canvas().waitForSelector( '.wp-block', { timeout: 120000 } );
```

Comment:
Waiting for the block selector doesn't give us real timing to the first block because the content flashes afterward, detaching all the nodes, and becomes actually editable much later (see the video below). Since the blocks load synchronously, IMO the time to first block is synonymous to the canvas becoming editable, which is correctly signalled by [the site tagline element](https://github.com/WordPress/gutenberg/pull/48138/files#r1113017842).


https://user-images.githubusercontent.com/1451471/220353001-e0629144-752b-4146-9d72-0c49e74dc8da.mov

Reply:
Site tag line is a dynamic block no? So it's a bit delayed? 
---

Can we instead fixing the flashes and only show the content of the editor when everything is ready?

Reply:
Actually, do you know the reasons for these flashes here? "first block" means first block for me, so `.wp-block` seems like a good selector for "any block" no?

Reply:
The issue is that the blocks are flashing, so if we wait for the `.wp-block` selector, we will get the timing **before the flash**, which doesn't indicate the correct load time, IMO. The tagline element doesn't flash and changes its state once the content becomes editable. I do agree, though, that fixing the flash would be a better solution here, as we could then reliably use the generic block selector. I don't know what causes the flash, but I can investigate. I'd appreciate any pointers, though, as I don't have any experience with the Site Editor internals.

Reply:
The flashing issue is caused by the `PostContent` block (responsible for rendering the entire post body). Depending on the [returned value from the `useCanEditEntity` hook](https://github.com/WordPress/gutenberg/blob/8486b48fc4994a0624368445cb2861bcc188a77e/packages/block-library/src/post-content/edit.js#L61), it renders either in the read-only or editable mode. The problem is that the `canEdit` value is initially `undefined`, so the content is always rendered in the read-only mode. Once the value becomes defined (boolean), it either stays in the read-only mode or switches to the editable mode, which is our case, and which ultimately causes the whole body to rerender. 

The fix seems to be pretty straightforward because all we need to do is to wait until the `useCanEditEntity` returns a boolean value, and then render the content. I've applied it in https://github.com/WordPress/gutenberg/pull/48138/commits/8486b48fc4994a0624368445cb2861bcc188a77e - please let me know if that makes sense to you @youknowriad. 

cc @t-hamano @griffbrad

Reply:
Here are page load demo videos before and after https://github.com/WordPress/gutenberg/commit/8486b48fc4994a0624368445cb2861bcc188a77e:

### Before

https://user-images.githubusercontent.com/1451471/222155102-5844bff7-c4e4-4a5e-a4fa-8f4180e0b32c.mov

### After

https://user-images.githubusercontent.com/1451471/222155065-7cbe8128-0326-43c7-a5ac-9c4ae384ab44.mov

Reply:
Great catch, that's an obvious improvement 👍 

The diff between `undefined` and `false` is kind of random here, maybe a better way would have been to actually check if the resolver to fetch the "rights" is done. But I think it's a good fix for now.

Reply:
> The diff between undefined and false is kind of random here, maybe a better way would have been to actually check if the resolver to fetch the "rights" is done. But I think it's a good fix for now.

Looks like waiting for a bool should be good! https://github.com/WordPress/gutenberg/blob/8dee3e6764da100b20ad9abd41af5df086b5f42f/docs/reference-guides/data/data-core.md#L48

Reply:
The fix: https://github.com/WordPress/gutenberg/pull/48642